js/index.js

```diff
@@ -70,24 +70,78 @@ function updateArticles() {
 
   function renderHTML(data) {
     var date = new Date();
+    var day = date.getDay();
     var hour = date.getHours();
-    var title, views;
-
-    var i = 1;
-    var counter = 0;
+    var minute = date.getMinutes();
+    switch("hour") {
+
+      // Uses the past day Json file and filters top articles from past 60 minutes.
+      case "hour":
+        var filtered = data.filter(function (a) {
+          var articleHour = a[2];
+          var articleMinute = a[3];
+          return (articleHour==hour-1 && articleMinute>=minute || articleHour==hour);
+        });
+        break;
+
+      // Uses the past 7 days Json file and filters top articles from past 24 hours.
+      case "day":
+
+        var filtered = data.filter(function (a) {
+          var articleHour = a[2];
+          var articleDay = a[3];
+          return (articleDay==day-1 && articleHour>=hour || articleDay==day);
```

Comment:
have a space on either side of ==, which should also be a ===
The difference between the 2 is that == will compare for equality AFTER performing type conversions (this is the same as type casting if you're familiar with the concept from java or C/C++). === will not perform type conversions, and thus is more strict but is a better indicator for equality.